hedera-node/src/main/java/com/hedera/services/txns/token/TokenUpdateTransitionLogic.java

```diff
@@ -147,8 +146,11 @@ private void transitionFor(TokenUpdateTransactionBody op) {
 				return;
 			}
 			if (!store.associationExists(newTreasury, id)) {
-				txnCtx.setStatus(INVALID_TREASURY_ACCOUNT_FOR_TOKEN);
-				return;
+				outcome = store.autoAssociate(newTreasury, id);
+				if (outcome != OK) {
+					txnCtx.setStatus(outcome);
```

Comment:
Probably it would be better to use `abortWith(outcome)` since `autoAssociate` attempts to make changes to the ledgers and in case of error, `abortWith(outcome)` reverts any pending changes to the ledgers.